cmd/kubectl-k8ssandra/register/registration.go

```diff
@@ -0,0 +1,143 @@
+package register
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8ssandra/k8ssandra-client/pkg/registration"
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/result"
+)
+
+type RegistrationExecutor struct {
+	DestinationName  string
+	SourceKubeconfig string
+	DestKubeconfig   string
+	SourceContext    string
+	DestContext      string
+	SourceNamespace  string
+	DestNamespace    string
+	ServiceAccount   string
+	Context          context.Context
+}
+
+func getDefaultSecret(saName, saNamespace string) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName + "-secret",
+			Namespace: saNamespace,
+			Annotations: map[string]string{
+				"kubernetes.io/service-account.name": saName,
+			},
+		},
+		Type: corev1.SecretTypeServiceAccountToken,
+	}
+}
+func getDefaultServiceAccount(saName, saNamespace string) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName,
+			Namespace: saNamespace,
+		},
+	}
+
+}
+
+func (e *RegistrationExecutor) RegisterCluster() result.ReconcileResult {
+	log.Printf("Registering cluster from %s Context: %s to %s Context: %s",
+		registration.GetKubeconfigFileLocation(e.SourceKubeconfig), e.SourceContext,
+		registration.GetKubeconfigFileLocation(e.DestKubeconfig), e.DestContext,
+	)
+	if e.SourceContext == e.DestContext && e.SourceKubeconfig == e.DestKubeconfig {
+		panic(errors.New("source and destination context and kubeconfig are the same, you should not register the same cluster to itself. Reference it by leaving the k8sContext field blank instead"))
```

Comment:
issue: No panics, handle errors.

Reply:
Again, this one is not handle-able, tis fatal my friend.

Reply:
No, this can be handled, this is part of validation. Just give user the same notification in prettier format.

We don't crash k8ssandra-operator when user gives an incorrect K8ssandraCluster definition, instead we log an error.

Reply:
A CLI app has a completely different lifecycle than a daemon, in the former case a faulty request should error out as early as possible, in the latter, the application should almost never exit.

Aside from anything else, this already provides the error to the user, and I'd rather supply the stack trace because it makes it clearer where the problem is coming from. Why hide this in an OSS piece of software when it can supply the user more information about what has gone wrong? 

Most frequently, I have difficulty finding the particular pathway an error has taken in our code because half of our errors look alike. That might be necessary in a long running app, but is just frustrating and unnecessary given the nature of this application.

Reply:
If you find error messages difficult to find, make better error messages, don't output stacktraces as a replacement for proper error messages. This is something the user has provided and as such could fix, it's not a bug where we could fix things. 

There's simply no reason not to return an error here and handle it correctly. panic is not a replacement for error. And why on earth are you even paniccing with an error? panic() takes a string as input, the entire string -> error -> string casting feels like a misunderstanding of what panic does.

Reply:
So you think that error messages should include the line number that they come from etc.? Does the logging framework we're using offer that as a feature? Have we turned it on?

Reply:
Why do you need a line number in a user error message?